### PR TITLE
Build the compiler with PCRE2

### PR DESCRIPTION
**Pull request by straight-shoota**
The explicit selection of PCRE2 via `-D use_pcre2` is intentional. We want the compiler to built with PCRE2 and not silently fall back to PCRE.

### CODE REVIEW

**Review by HertzDevil — COMMENTED** [review]
The first generation cross-compilation command to Windows also needs an explicit `-Duse_pcre2`

File: .github/workflows/win.yml [hunk 1]
```diff
@@ -42,6 +42,7 @@ jobs:
         with:
           path: | # openssl and llvm take much longer to build so they are cached separately
             libs/pcre.lib
+            libs/pcre2.lib
```
Comment (HertzDevil) [inline .github/workflows/win.yml:45]:
```suggestion
            libs/pcre2-8.lib
```

(Also I didn't notice this wasn't cached before...)

File: .github/workflows/win.yml [hunk 2]
```diff
@@ -332,7 +333,7 @@ jobs:
         run: make -f Makefile.win deps
       - name: Link Crystal executable
         run: |
-          Invoke-Expression "cl crystal.obj /Fecrystal src\llvm\ext\llvm_ext.obj $(llvm\bin\llvm-config.exe --libs) libs\pcre.lib libs\gc.lib WS2_32.lib advapi32.lib libcmt.lib dbghelp.lib ole32.lib shell32.lib legacy_stdio_definitions.lib /link /LIBPATH:$(pwd)\libs /STACK:0x800000 /ENTRY:wmainCRTStartup"
+          Invoke-Expression "cl crystal.obj /Fecrystal src\llvm\ext\llvm_ext.obj $(llvm\bin\llvm-config.exe --libs) libs\pcre2.lib libs\gc.lib WS2_32.lib advapi32.lib libcmt.lib dbghelp.lib ole32.lib shell32.lib legacy_stdio_definitions.lib /link /LIBPATH:$(pwd)\libs /STACK:0x800000 /ENTRY:wmainCRTStartup"
```
Comment (HertzDevil) [inline .github/workflows/win.yml:336]:
```suggestion
          Invoke-Expression "cl crystal.obj /Fecrystal src\llvm\ext\llvm_ext.obj $(llvm\bin\llvm-config.exe --libs) libs\pcre2-8.lib libs\gc.lib WS2_32.lib advapi32.lib libcmt.lib dbghelp.lib ole32.lib shell32.lib legacy_stdio_definitions.lib /link /LIBPATH:$(pwd)\libs /STACK:0x800000 /ENTRY:wmainCRTStartup"
```

**Comment by straight-shoota** [comment]
Converted to draft. There are a couple of CI issues with PCRE2 that need to be resolved first.

**Comment by straight-shoota** [comment]
CI is all green. This is ready for review now.

**Review by HertzDevil — APPROVED** [review]

File: .github/workflows/linux.yml [hunk 3]
```diff
@@ -15,13 +15,18 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        crystal_bootstrap_version: [1.2.2, 1.3.2, 1.4.1, 1.5.1, 1.6.2, 1.7.2]
+        crystal_bootstrap_version: [1.2.2, 1.3.2, 1.4.1, 1.5.1, 1.6.2]
+        env: ["USE_PCRE1=true"]
```
Comment (Sija) [inline .github/workflows/linux.yml:19]: Hash notation seems more natural:

```suggestion
        env: 
          USE_PCRE1: true
```

File: .github/workflows/linux.yml [hunk 4]
```diff
@@ -15,13 +15,18 @@ jobs:
...
         include:
           # libffi is only available starting from the 1.2.2 build images
           - crystal_bootstrap_version: 1.0.0
             flags: -Dwithout_ffi
+            env: USE_PCRE1=true
```
Comment (Sija) [inline .github/workflows/linux.yml:24]:
```suggestion
            env:
              USE_PCRE1: true
```

File: .github/workflows/linux.yml [hunk 5]
```diff
@@ -15,13 +15,18 @@ jobs:
...
           - crystal_bootstrap_version: 1.1.1
             flags: -Dwithout_ffi
+            env: USE_PCRE1=true
```
Comment (Sija) [inline .github/workflows/linux.yml:27]:
```suggestion
            env:
              USE_PCRE1: true
```

File: .github/workflows/linux.yml [hunk 6]
```diff
@@ -33,7 +38,7 @@ jobs:
         run: bin/ci prepare_build
 
       - name: Test
-        run: FLAGS=${{ matrix.flags }} bin/ci build
+        run: FLAGS=${{ matrix.flags }} ${{ matrix.env}} bin/ci build
```
Comment (Sija) [inline .github/workflows/linux.yml:41]:
```suggestion
        run: FLAGS=${{ matrix.flags }} ${{ matrix.env }} bin/ci build
```

File: .github/workflows/linux.yml (hunk 3 above)
Comment (Sija) [reply .github/workflows/linux.yml]: Oh shit, I've noticed just now that's not `jobs.<job_id>.env` but rather a matrix variable called `env`. In this case using hash notation rather won't work, sorry for the mixup. btw, I'd suggest renaming this variable to some more descriptive, like `flags`, so it gives better understanding what is it for and won't mixup with the `env` key, which is _a thing_ for GHA.

**Review by sdogruyol — APPROVED** [review]

File: .github/workflows/linux.yml (hunk 3 above)
Comment (Sija) [reply .github/workflows/linux.yml]: @straight-shoota This most likely broke the CI :(

Comment (straight-shoota) [reply .github/workflows/linux.yml]: Oh shit. Thanks for noticing.
What's really, really bad about this is that the GitHub Actions summary on the PR and commit doesn't mention this. The workflow file failed to parse but that's not reported as a failure? WTF.

**Pull request merged**